app/controllers/comments_controller.rb

```diff
@@ -1,32 +1,31 @@
 class CommentsController < ApplicationController
-  before_action :find_article, only: %i[create]
   before_action :find_comment, only: %i[destroy]
-
   http_basic_authenticate_with name: 'user', password: 'secret', only: :destroy
 
   def create
-    @comment = @article.comments.create(comments_params)
-    redirect_to article_path(@article)
+    @comment = Comment.new(comments_params)
+    @comment.article_id = params[:article_id]
+    begin
+      @comment.save!
+    rescue ActiveRecord::RecordInvalid
+    end
+    redirect_back fallback_location: articles_url
   end
 
   def destroy
-    if @comment.article_id == params[:article_id].to_i
-      @comment.destroy
-    end
+    @comment.destroy if @comment.article_id.eql? params[:article_id].to_i
```

Comment:
You can find the `@comment` using its `id` and the `article_id` in the same query. That way you don't need to do this comparison here.

Reply:
I made this query:
`@comment = Comment.find_by(id: params[:id], article_id: params[:article_id])`

Reply:
if returns `nil` i made a ternary conditional:
`@comment.nil? ? not_found : @comment`

Reply:
Here, this method assigns the found comment to the variable `@comment`, so it is not necessary to return it. Consider using the`if`modifier like this:

`not_found if @comment.nil?`

Reply:
Got it! also, i will keep  in mind this little things